Google.Api.Gax.Grpc/CallSettings.cs

```diff
@@ -325,6 +327,14 @@ public static CallSettings FromFieldMask(string fieldMask)
         internal static CallSettings FromGoogleRequestParamsHeader(string parameterName, string value) =>
             FromHeader(RequestParamsHeader, parameterName + "=" + Uri.EscapeDataString(value ?? ""));
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="paramNameValues"></param>
+        /// <returns></returns>
+        internal static CallSettings FromGoogleRequestParamsHeader(Dictionary<string, string> paramNameValues) =>
+            FromHeader(RequestParamsHeader, string.Join("&", paramNameValues.Select(nameVal => nameVal.Key + "=" + Uri.EscapeDataString(nameVal.Value ?? ""))));
```

Comment:
Is the order important here? Would it make sense to take a list of key/value pairs instead? (Potentially even just a list of values that is formatted by the HeaderParameterExtractor?)